### ISSUE TITLE
Stop using tables for doc list

### ISSUE THREAD

**Issue by boonebgorges** [boonebgorges/buddypress-docs#466]
See #408.

The new breadcrumbs in 1.9 are forcing the issue of the table markup of doc lists. Let's ditch them.

I really like the look of what @JonathanReeve has suggested in #408. I'm going to see if I can merge it in a way that makes sense.

I will probably leave the existing docs-loop.php in place for backward compatibility. Sites will get the new view by default, but will be able to keep the old one with a filter.

I'm going to use this ticket to track the process.

**Comment by dcavins** [comment]
This would be a good chance to update the css approach of the default bp-docs.css too, to minimize id selectors.

**Comment by boonebgorges** [comment]
Can you say more about this? What's the problem with the current selectors?

**Comment by dcavins** [comment]
Some of the selectors have a nuclear-level amount of selector juice, so overriding them in a theme requires even more juice and results in themes having to write a lot of specific styles. Like:
`#buddypress #bp-docs-all-docs li`
`#associated_group_summary .item`

Part of that I'm sure has to do with overriding selectors that are part of bp-default. But if we can rely on classes (and the fewer the better) for selecting elements in the basic css file, it makes it more likely that a theme's element CSS styles will be respected. For instance, if a theme has added styling to `.button`, it'll get steamrolled by BP Docs' much more specific selector, then the themer will have to duplicate the `.button` style but with more selector juice. Ideally, we'd apply styles to `.button` also but allow a theme sheet's later load to provide the override.

Making a default theme that is both compelling and written in such a way as to allow easy/logical overriding and inheritance is a really hard job. :)

**Comment by boonebgorges** [comment]
Gotcha. I'm not sure how much of that will be touched just by 
eliminating the table. I don't relish the thought of rewriting all the 
markup in the whole plugin :) But I'll keep it in mind as I poke around.

**Comment by dcavins** [comment]
Oh we definitely don't want to rewrite any more of the markup than we have to to preserve styling that people have sweated over. But, adding classes where appropriate would be a good improvement.

**Comment by boonebgorges** [comment]
Hey @dcavins - I've done some initial work on the `notables` branch. If you have time, would you mind pulling the changes and having a look? I'd love to get your initial impressions of the changes in markup and styling (which are based largely on @JonathanReeve's work)

**Comment by dcavins** [comment]
Hi Boone-

I'm a fan. I like especially how info is added when appropriate (when a doc is in a folder or associated with a group), but doesn't take up space or add clutter when it's not.

I made a few changes as I was playing with the layout and checking it in the various WP-default themes. You can see my branch here: https://github.com/dcavins/buddypress-docs/commits/no-tables-edits

In general, this goes a long way toward simplifying the css selectors & applying classes where appropriate. I think that this layout will be much easier to customize because of improved selectors but also because of the addition of useful filter/action points in the doc loop.

One question I do have is that the result is a series of `<div>`s in order. Would it make sense for these objects to be list items in an unordered list? (Would that distinction help some types of users navigate the page better, like screen-reader users? Screen readers announce "list of 7 items" which seems like it would be helpful in a list of docs.)

Thanks for asking me to take a look.

**Comment by boonebgorges** [comment]
Awesome - thanks for looking, @dcavins. I'll take some time in the 
upcoming days to check out your mods.

Regarding the div-fest: I generally prefer to use ULs etc, but here I 
was coping with Jonathan did, and really, what WordPress themes do in 
general. If you are on the side of the ULs, then that's enough to sway me.

I'm going to go ahead with these changes as the default interface for 
1.9, and we'll see how many people I can manage to tick off.

**Comment by dcavins** [comment]
Hi Boone-

I was working with the no-tables branch again and thought that I'd check that the "use the legacy directory view" filter worked as advertised. It looks like it needs to be applied in a few more places. See https://github.com/dcavins/buddypress-docs/commit/4cdf1e25cef4172e7015441beeeb7008382fd63a.

I'm not sure about using a `ul` over successive `div`s. I like the idea of it, but you're right that WP presents archives as `article`s or `div`s so maybe it makes sense to go with the more common structure, even though I like the idea of a `ul`. It's probably more important to work with more themes' base styles than to do this the "right" way--and there's no consensus on that anyway.

**Comment by boonebgorges** [comment]
Good catches, @dcavins. The fact that this file is called in a bunch of places makes me wonder whether it's worth putting the filter flag into `bp_docs_locate_template()` itself. So you'd always call `bp_docs_locate_template( 'docs-loop.php' )`. I guess I can imagine cases where this might be a bit confusing to devs, and it does create a problem with the `bp_buffer_template_part()` usage, as you'd have to introduce some magic into BP's template selection logic too. Maybe this is not a great idea after all :) What do you think?

**Comment by dcavins** [comment]
Hi Boone-

I'm not sure what the answer is. I recently had to do some fairly extensive modification to the templates (and started by trying to override template parts but ended up being able to do everything via filters and actions, which is awesome, so thanks for all of those hooks), and also wondered if we couldn't simplify the whole thing down to running all template calls through a single routing function like `bp_docs_locate_template()`. But I know it's complicated by the pre-compat/post-compat thing, and I know that I don't have a solid overview of how everything works.

As I was working on making the table view work, here are some things I noted:
* `integration-groups.php` seems to duplicate the `query-builder.php` logic--could we combine?
* I could never figure out when `docs/index.php` is called, maybe only with pre-1.7 BP?

Do you remember enough about the historical development of bp-docs to shine some light on how the various pieces work together? Templating is hard. :)

Thanks for taking the time to respond. Best of luck with the move!

**Comment by boonebgorges** [comment]
>  But I know it's complicated by the pre-compat/post-compat thing, and I know that I don't have a solid overview of how everything works.

Yeah. The more I think about this, the more I think it's probably best not to overarchitect. Putting the filter in multiple places like you've done (and maybe wrapping the check in a function `bp_docs_use_legacy_template()` or something like that) is straightforward enough.

> integration-groups.php seems to duplicate the query-builder.php logic--could we combine?

Yes, I think so. I don't remember the history here, but I don't see any reason why the logic can't be combined.

> I could never figure out when docs/index.php is called, maybe only with pre-1.7 BP?

Yes, it's only called when not using theme compatibility. It's a top-level template (includes header, footer, etc).
